spd/identity_insertion.py

```
@@ -32,15 +33,28 @@ def pre_id_hook(
     return (mod.pre_identity(args[0]),), {}
 
 
-def insert_identity_operations_(target_model: nn.Module, identity_patterns: list[str]) -> None:
+def insert_identity_operations_(
+    target_model: nn.Module, identity_module_info: list[ModulePatternInfoConfig]
+) -> None:
     """Insert identity layers before specified modules.
 
     Args:
         target_model: The model to modify
-        identity_patterns: Patterns matching modules to prepend identity ops to
+        identity_module_info: List of ModulePatternInfoConfig. The C values are ignored here
+            (used later when creating components), only patterns are used for matching.
     """
+    # Extract just the patterns (ignore C values for insertion)
+    identity_module_paths: list[str] = []
+    matched_patterns: set[str] = set()
+    for info in identity_module_info:
+        for name, _ in target_model.named_modules():
+            if fnmatch.fnmatch(name, info.module_pattern):
+                matched_patterns.add(info.module_pattern)
```

I think you should raise an error if the info.module_pattern is already in matched_patterns, just as you do for the target module patterns.